lib/test-definition.rb

```diff
@@ -284,11 +285,23 @@ def run(deployment, transport, iteration)
       verify_snmp_bono_latency if ENV['BONO_SNMP'] == "Y"
     ensure
       retval &= cleanup
+
+      # If the test failed and we have retries set, recursively call run
+      if !retval and @repeat_on_failure > 0
+        @repeat_on_failure -= 1
+        puts "WARNING - Test failed iteration, retrying"
+        retval = self.run(deployment, transport, iteration)
+      end
+
       TestDefinition.unset_current_test
     end
     return retval
   end
 
+  def set_repeat_on_failure(count)
```

Comment:
The `attr_writer`section at the start of the class frees you from having to manually write accessor methods.